### ISSUE TITLE
Delete collection that was actually an alias causes crash

### ISSUE THREAD

**Issue by ryansorensen** [typesense/typesense#223]
## Description

The "drop a collection" function can (accidentally) be run against an alias. This appears to cause some kind of corruption where the service crashes with error `terminate called after throwing an instance of 'std::bad_alloc'` and can only be fixed by deleting the data directory. 

## Steps to reproduce

1. Create collection `assets_1`
2. Assign alias `assets` to collection `assets_1`
3. Using the collections endpoint, delete a collection named `assets`

```
# Create collection `assets_1`
$ curl --location --request POST 'http://localhost:8108/collections' \
--header 'X-TYPESENSE-API-KEY: pass' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "assets_1",
    "default_sorting_field": "asset_id",
    "fields": [
        {
            "name": "asset_id",
            "type": "int32"
        }
    ]
}'
```
```
# Assign alias `assets` to collection `assets_1`
$ curl --location --request PUT 'http://localhost:8108/aliases/assets' \
--header 'X-TYPESENSE-API-KEY: pass' \
--header 'Content-Type: application/javascript' \
--data-raw '{
    "collection_name": "assets_1"
}'
```
**I believe the step below should throw an error, as the user should not be able to delete a collection from an alias**
```
# Delete the alias as if it were a collection
$ curl --location --request DELETE 'http://localhost:8108/collections/assets' \
--header 'X-TYPESENSE-API-KEY: pass'

> DELETE /collections/assets HTTP/1.1
> Host: localhost:9114
> User-Agent: curl/7.64.1
> Accept: */*
> X-TYPESENSE-API-KEY: pass
>
< HTTP/1.1 200 OK
< Connection: keep-alive
< content-type: application/json; charset=utf-8
< transfer-encoding: chunked
<
* Connection #0 to host localhost left intact
{"created_at":1613791437,"default_sorting_field":"asset_id","fields":[{"facet":false,"name":"asset_id","optional":false,"type":"int32"}],"name":"assets_1","num_documents":0,"num_memory_shards":4}* Closing connection 0

```

Any further attempts to interact with the system cause it to crash, and it cannot be started until the data directory is deleted

```
# attempt to list collections
$ curl --location --request GET 'http://localhost:8108/collections' \
--header 'X-TYPESENSE-API-KEY: pass'

...
Empty reply from server
...
connect to ::1 port 8108 failed: Connection refused
```

## Expected Behavior

"Drop collection" function should throw an error if the argument is an alias and not a collection

## Actual Behavior

"Drop collection" function does delete the real collection (the target of the alias) but then the service becomes unstable and will not perform any functions. 

## Metadata

**Typsense Version**:
image: typesense/typesense:0.19.0

**OS**:
Docker 20.10.2 running on Mac OS Big Sur

**Comment by kishorenc** [comment]
Yikes, should be using the resolved collection name during deletion. I already have a fix for this that allows the deletion but handles everything properly using the actual collection name. I will produce a patched build for you to test in a couple of days. Thanks for reporting this issue @ryansorensen!

**Comment by ryansorensen** [comment]
Amazingly fast resolution! Happy to test and report back.

**Comment by kishorenc** [comment]
Sorry for the delay @ryansorensen -- please test against `typesense/typesense:0.20.0.rc11`.

**Comment by kishorenc** [comment]
Addressed as part of `v0.20.0` released last week.
